src/query_plan/operation.rs

```diff
@@ -612,13 +897,6 @@ impl NormalizedSelectionSet {
 }
 
 impl NormalizedFieldSelection {
```

Comment:
since you created separate modules for each selection -> should those impls move to the corresponding module space?

Reply:
@dariuszkuc 
Some of the `impl`s contain algorithms that shouldn't use the underlying struct fields, and should instead use the field's methods. To get that behavior, the `impl`s are outside the files. (We could alternatively have a bunch of top-level functions instead of `impl`s, but that feels a bit disorganized.)